### PR TITLE
Add ignoreList  to GHIndexer

### PR DESCRIPTION
**Pull request by RiadGahlouz**
This PR adds an ignore list to the github indexer job to be able to filter out unwanted repositories from being indexed.

### CODE REVIEW

**Comment by dnfgituser** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/NuGet/NuGet.Jobs?pullRequest=1005) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/NuGet/NuGet.Jobs?pullRequest=1005) before we can accept your contribution.<br/><br/>:x: RiadGahlouz [sign now](https://cla.dotnetfoundation.org/NuGet/NuGet.Jobs?pullRequest=1005)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/NuGet/NuGet.Jobs?pullRequest=1005) it.</sub>

File: src/NuGet.Jobs.GitHubIndexer/GitHubIndexerConfiguration.cs [hunk 1]
```diff
@@ -46,5 +47,10 @@ public class GitHubIndexerConfiguration
         /// How long to sleep after succeeding. If the job fails this setting will be ignored.
         /// </summary>
         public TimeSpan SleepAfterSuccess { get; set; }
+
+        /// <summary>
+        /// The list of repositories to be ignored by the indexing job. ("{repoOwner}/{repoName}")
+        /// </summary>
+        public HashSet<string> IgnoreList {  get; set; }
```
Comment (joelverhagen) [inline src/NuGet.Jobs.GitHubIndexer/GitHubIndexerConfiguration.cs:54]: Case sensitivity?

File: src/NuGet.Jobs.GitHubIndexer/GitHubIndexerConfiguration.cs [hunk 2]
```diff
@@ -46,5 +47,10 @@ public class GitHubIndexerConfiguration
         /// How long to sleep after succeeding. If the job fails this setting will be ignored.
         /// </summary>
         public TimeSpan SleepAfterSuccess { get; set; }
+
+        /// <summary>
+        /// The list of repositories to be ignored by the indexing job. ("{repoOwner}/{repoName}")
+        /// </summary>
```
Comment (loic-sharma) [inline src/NuGet.Jobs.GitHubIndexer/GitHubIndexerConfiguration.cs:53]: There's an extra space below in `{  get; set; }

File: src/NuGet.Jobs.GitHubIndexer/GitRepoSearchers/GitHub/GitHubSearcher.cs [hunk 3]
```diff
@@ -135,10 +135,12 @@ private async Task<IReadOnlyList<WritableRepositoryInformation>> GetResultsFromG
                     }
 
                     // TODO: Block unwanted repos (https://github.com/NuGet/NuGetGallery/issues/7298)
```
Comment (loic-sharma) [inline src/NuGet.Jobs.GitHubIndexer/GitRepoSearchers/GitHub/GitHubSearcher.cs:137]: Can we remove this line? :)

File: src/NuGet.Jobs.GitHubIndexer/GitRepoSearchers/GitHub/GitHubSearcher.cs [hunk 4]
```diff
@@ -135,10 +135,12 @@ private async Task<IReadOnlyList<WritableRepositoryInformation>> GetResultsFromG
...
-                    resultList.AddRange(response);
+                    var ignoreList = _configuration.Value.IgnoreList;
+                    var filteredRepoList = response.Where(x => !ignoreList.Contains(x.Id, StringComparer.OrdinalIgnoreCase));
```
Comment (joelverhagen) [inline src/NuGet.Jobs.GitHubIndexer/GitRepoSearchers/GitHub/GitHubSearcher.cs:139]: Maybe log the intersection of the two sets: these are the actually ignored repo names

**Review by joelverhagen — APPROVED** [review]

**Review by loic-sharma — APPROVED** [review]

**Review by ryuyu — APPROVED** [review]

**Pull request merged**